user/views.py

```diff
@@ -0,0 +1,70 @@
+import json
+import bcrypt
+import jwt
+import re
+
+from django.views import View
+from django.http  import JsonResponse
+
+from user.models  import User
+from my_settings  import SECRET_KEY
+
+class UserCheckEmailView(View):
+    def post(self, request):
+        data  = json.loads(request.body)
+        email = data.get('email')
+
+        try:
+            email_regex = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+
+            if not email:
+                return JsonResponse({'message': 'KEY_ERROR'}, status=400)
+            if not email_regex.match(email):
+                return JsonResponse({'message': 'INVALID_EMAIL'}, status=400)
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message': 'GO_SINGIN'}, status=200)
+            if not User.objects.filter(email=email).exists():
+                return JsonResponse({'message': 'GO_SINGUP'}, status=200)
+        
+        except:
+            return JsonResponse({'message': 'KEY_ERROR'}, status=400)
+    
+class UserSignUpView(View):
+    def post(self, request):
+        data       = json.loads(request.body)
+        email      = data.get('email')
+        password   = data.get('password')
+        last_name  = data.get('last_name')
+        first_name = data.get('first_name')
+
+        password_regex = re.compile('^(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{6,}$')
+        if not password_regex.match(password):
+            return JsonResponse({'message': 'INVALID_PASSWORD'}, status=400)
+        encoded_pw = password.encode('utf-8')
+        hashed_pw  = bcrypt.hashpw(encoded_pw, bcrypt.gensalt())
+        encrypt_pw  = hashed_pw.decode('utf-8')
+        User.objects.get_or_create(
+            email      = email,
+            password   = encrypt_pw,
+            last_name  = last_name,
+            first_name = first_name
+            )
+        return JsonResponse({'message': 'SIGN_UP_SUCCESS'}, status=200)
```

Comment:
```suggestion
        password_regex = re.compile('^(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{6,}$')
        if not password_regex.match(password):
            return JsonResponse({'message': 'INVALID_PASSWORD'}, status=400)
            
        encoded_pw = password.encode('utf-8')
        hashed_pw  = bcrypt.hashpw(encoded_pw, bcrypt.gensalt())
        encrypt_pw  = hashed_pw.decode('utf-8')
        
        User.objects.get_or_create(
            email      = email,
            password   = encrypt_pw,
            last_name  = last_name,
            first_name = first_name
            )
        return JsonResponse({'message': 'SIGN_UP_SUCCESS'}, status=200)
```
로직이 달라지고 있기 때문에 빈 줄을 삽입해주시는 것이 좋습니다

Reply:
수정 진행하였습니다. 감사합니다.